java/pom.xml

```diff
@@ -251,7 +251,7 @@
       <plugin>
         <groupId>pl.project13.maven</groupId>
         <artifactId>git-commit-id-plugin</artifactId>
-        <version>2.2.2</version>
+        <version>4.0.5</version>
```

Comment:
This plugin was causing problems with shallow copies, it should have been able to handle it but it looks like our version was too old. The new version introduces the cli switch I added to `java_build.sh`. This is now the newest possible version we can use as 5.0.0 does not support java < 11.